### PR TITLE
Fix JS error when clicking GPay button from blocks checkout with subscription in cart.

### LINKED ISSUE #8499: Error on JN site when using Google Pay with subscription
### Describe the bug
The below behavior happens when attempting to pay for a subscription on Blocks checkout and in a JN store. We weren't able to reproduce the same behavior on a local setup.

image

### To Reproduce
* Create a subscription with a sign-up fee
* Log out from the store
* Add subscription to the cart, navigate to Blocks checkout
* Click GPay button and observe the error

### Additional context
From a very quick scan, the error seems to be coming from https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-request/blocks/use-initialization.js#L88.

### PR DESCRIPTION
**Pull request by brettshumaker**
Fixes #8499

#### Changes proposed in this Pull Request
This PR passes along the click event to the `onButtonClick()` function when the GPay PRB button is clicked.

#### Testing instructions

1. Go to a JN store.
2. Create a subscription with a sign-up fee
3. Log out from the store
4. Add subscription to the cart, navigate to Blocks checkout
5. Click GPay button and see a browser alert that you need to log in or create an account on the site before you can complete the transaction.

You should see something like this after clicking the GPay button:
screenshot_2024-04-04_at_11 47 38___am_720

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8579` or branch name `8499/fix-js-error-on-gpay-click` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=8499/fix-js-error-on-gpay-click) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f220d18c3b4d631afe3f6c00758ec7c31468c4c7
- Build time: 2024-04-05 13:04:35 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-request/blocks/payment-request-express.js [hunk 1]
```diff
@@ -75,8 +75,8 @@ const PaymentRequestExpressComponent = ( {
 		onPaymentRequestAvailable( paymentRequestType );
 	} );
 
-	const onPaymentRequestButtonClick = () => {
-		onButtonClick();
+	const onPaymentRequestButtonClick = ( evt ) => {
+		onButtonClick( evt, paymentRequest );
```
Comment (FangedParakeet) [inline client/payment-request/blocks/payment-request-express.js]:
```suggestion
	const onPaymentRequestButtonClick = ( event ) => {
		onButtonClick( event, paymentRequest );
```
> [Variable and function names should be full words, using camel case with a lowercase first letter.](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#naming-conventions)

🤷

**Review by FangedParakeet — CHANGES_REQUESTED** [review]
One teeny suggestion, but this change looks fine and works just as advertised. 🎉

File: client/payment-request/blocks/payment-request-express.js (hunk 1 above)
Comment (brettshumaker) [reply client/payment-request/blocks/payment-request-express.js]: Fair enough. I was matching the variable name used [here](https://github.com/Automattic/woocommerce-payments/blob/4fa44519a2e5da4b723c01e3ba92f513790a4db2/client/payment-request/blocks/use-initialization.js#L85) but happy to change it. 😃

Comment (FangedParakeet) [reply client/payment-request/blocks/payment-request-express.js]: Yeah, I know, I didn't like that var there either, but I can't moan at you for that one in this PR. 😂

**Review by FangedParakeet — APPROVED** [review]
LGTM. 🚢

**Pull request merged**